fsi-workflow/01_solidMesh/tasks.md

```diff
@@ -1,182 +1,147 @@
+![FreeCAD: Mesh group](./images/Groups02.png)
 
- - repeat for tip surface (it will be used for single physics test)
- - rename the group to **tip**
+  In case you cannot select anything and you get an error "Active Task Dialog found!", you might need to switch to the `Tasks` tab and click `OK` or `Cancel` to exit from the previous task.
 
-Select the mesh->double click-> press apply to remesh and create groups-> ok
+- In the `Model` tab, there should now also be a `MeshGroup` under the `FEMMeshGmsh`.
 
-==IMPORTANT==: you need to remesh otherwise the groups won't be created
+![FreeCAD: Added MeshGroup](./images/Groups03.png)
 
-## Save INP file and verify
+- In the `Tasks` tab, select `Label` as `Identifier used for mesh export` and `Face, Edge, Vertex` as `Selection mode`:
 
- - keep the mesh selected
- - File->export 
- - name the file  **wing2312.inp**
- - save
- - close
- - File->Save the FreeCAD model
+![FreeCAD: Add surface to mesh group](./images/Groups04.png)
 
-open the **inp** file with a text editor:
+- Click `Add`, then click on the rendering to select the profile of the wing (pay attention to reference frame to identify it), and `OK` to add the surface to the mesh group. You can rotate the view using the cude on the upper right corner, or [using your mouse](https://wiki.freecad.org/Mouse_navigation) (e.g., by `Shift` + `right click`).
+- In the `Model` tab, select the `MeshGroup` and rename its `Label` to `root`.
 
+![FreeCAD: Rename mesh group](./images/Groups05.png)
 
- - look for **ELSET**, skip Evolume, you may remove remove all other ELSET
- - look for **NSET**, skip Nall
-     - *NSET, NSET=tip_Nodes 
-     - *NSET, NSET=wetSurface_Nodes
-     - *NSET, NSET=root_Nodes
+- Repeat for other surfaces (they are 4: `upper`, `lower`, `tip` and `trailing`. Pay attention to the trailing edge surface, you need to zoom-in to see it). Remember to first select `FEMMeshGmsh`.
+- Rename the group (**TODO: Which? How?**) to `wetSurface`
+
+- Repeat for the leading edge (it will be used for single physics tests), with group name `tip`.
+
+**TODO:** Add screenshot with final setup
+
+Select the mesh -> double click -> click `Apply` to remesh and create groups -> click `OK`.
+
+**IMPORTANT**: You need to remesh otherwise the groups won't be created.
+
+## Export the mesh file and verify
+
+- Keep the mesh selected.
+- Click `File` -> `Export...`.
+- Name the file  `wing2312.inp`.
+- Select `FEM mesh formats (*.dat, *.inp, ...)`.
+- Click `Save`.
+
+Save also the FreeCAD model with `File` -> `Save`.
+
+To verify, open the `wing2312.inp` file you just generated with a text editor:
+
+- Look for `ELSET`: Skip the `Evolume` set, you may remove all other `ELSET` entries (but you don't need to, they are just not used in the rest of the course)
+- Look for `NSET`: skip `Nall`
+  - `*NSET, NSET=tip_Nodes` 
+  - `*NSET, NSET=wetSurface_Nodes`
+  - `*NSET, NSET=root_Nodes`
 - Notice the names of all the sets of nodes for each of the groups.
```

Comment:
What should I notice here?

Reply:
the names of the sets, "tip_Nodes", "root_Nodes" etc. Those names are the ones that you have to insert in the CalculiX file when referring to the boundaries and loads.

Reply:
@Ccaccia73 I executed all the instructions again, and everything went smooth, thanks for the updates! :+1:

However, in the `.inp` file I export, I get:

```
*Node, NSET=Nall
** GroupID: 1 --> GroupName: root_Nodes --> GroupElementType: Node
*NSET, NSET=root_Nodes
** GroupID: 3 --> GroupName: MeshGroup002_Nodes --> GroupElementType: Node
*NSET, NSET=MeshGroup002_Nodes
** GroupID: 4 --> GroupName: MeshGroup001_Nodes --> GroupElementType: Node
*NSET, NSET=MeshGroup001_Nodes
```

(I tried a few times to create the correct groups, that's why the order is mixed up)

I am wondering why you get `wetSurface_Nodes` and `tip_Nodes`, but I don't. I have named the groups in FreeCAD:

![Screenshot from 2024-08-06 12-29-28](https://github.com/user-attachments/assets/3a7fb59b-17eb-4f1c-9727-c56bd9f04d12)

Reply:
As to the group order, I believe it is "random", or sort of. In my tests, I didn't always have the same order. But that should not be an issue. As to the names, did you remesh after renaming?

Reply:
Yes, I did remesh after renaming.

Reply:
Once it happened to me, too. But I didn't understand how to solve this. In the end I deleted the group and made it again. I don't know if it is a bug or there is a good way and a bad way to create groups with labels. (the name that you see is the group name, not the label). We'll see. BTW have you seen the 4 figures for the wetSurface group? they were indended to be smaller and in a 2x2 table, but it didn't work

Reply:
Tried again, it worked, no clue what I might have done wrong before.